src/OracleService/Protocols/OracleHttpsProtocol.cs

```diff
@@ -62,6 +64,9 @@ public async Task<(OracleResponseCode, string)> ProcessAsync(Uri uri, Cancellati
                     message = await client.GetAsync(uri, HttpCompletionOption.ResponseContentRead, cancellation);
                     if (message.Headers.Location is not null)
                     {
+                        if (stopwatch.Elapsed > client.Timeout)
```

Comment:
How many requests can be made within this time frame?

Reply:
```C#
                for(int i = 0; i<5 && message == null; i++)
                {
                    if (!Settings.Default.AllowPrivateHost)
                    {
                        IPHostEntry entry = await Dns.GetHostEntryAsync(uri.Host, cancellation);
                        if (entry.IsInternal())
                            return (OracleResponseCode.Forbidden, null);
                    }
                    message = await client.GetAsync(uri, HttpCompletionOption.ResponseContentRead, cancellation);
                    if (message.Headers.Location is not null)
                    {
                        if (stopwatch.Elapsed > client.Timeout)
                            return (OracleResponseCode.Timeout, null);


                        uri = message.Headers.Location;
                        message = null;
                    }
                };
```

Reply:
> How many requests can be made within this time frame?

It's the same time as one request

Reply:
Agree with @roman-khimov